Loop/Managers/Alerts/AlertManager.swift

```diff
@@ -156,75 +155,24 @@ extension AlertManager {
-        // Assume if it was delivered, the trigger should be .immediate.
-        playbackAnyNotificationRequest(notification.request, usingTrigger: .immediate)
-    }
-
-    private func playbackPendingNotificationRequest(_ request: UNNotificationRequest) {
-        playbackAnyNotificationRequest(request)
-    }
-    
-    private func playbackAnyNotificationRequest(_ request: UNNotificationRequest, usingTrigger trigger: Alert.Trigger? = nil) {
-        guard let savedAlertString = request.content.userInfo[AlertUserNotificationUserInfoKey.alert.rawValue] as? String,
-            let savedAlertTimestampString = request.content.userInfo[AlertUserNotificationUserInfoKey.alertTimestamp.rawValue] as? String,
-            let savedAlertTimestamp = AlertManager.timestampFormatter.date(from: savedAlertTimestampString) else  {
-            self.log.error("Could not find persistent alert in notification")
-            return
-        }
-        do {
-            let savedAlert = try Alert.decode(from: savedAlertString)
-            let newTrigger = trigger ?? determineNewTrigger(from: savedAlert, timestamp: savedAlertTimestamp)
-            let newAlert = Alert(identifier: savedAlert.identifier,
-                                 foregroundContent: savedAlert.foregroundContent,
-                                 backgroundContent: savedAlert.backgroundContent,
-                                 trigger: newTrigger,
-                                 sound: savedAlert.sound)
-            self.log.debug("Replaying %@Alert: %@ with %@trigger %@",
-                           trigger != nil ? "" : "Pending ",
-                           trigger != nil ? "" : "new ",
-                           savedAlertString, "\(newTrigger)")
-            self.replayAlert(newAlert)
-        } catch {
-            self.log.error("Could not decode alert: error %@, from %@", error.localizedDescription, savedAlertString)
-        }
+    private func playbackAlertsFromPersistence() {
+        playbackAlertsFromAlertStore()
     }
     
-    private func determineNewTrigger(from alert: Alert, timestamp: Date) -> Alert.Trigger {
-        switch alert.trigger {
-        case .immediate:
-            return alert.trigger
-        case .delayed(let interval):
-            let triggerTime = timestamp.addingTimeInterval(interval)
-            let timeIntervalSinceNow = triggerTime.timeIntervalSinceNow
-            if timeIntervalSinceNow < 0 {
-                // Trigger time has passed...trigger immediately
-                return .immediate
-            } else {
-                return .delayed(interval: timeIntervalSinceNow)
+    private func playbackAlertsFromAlertStore() {
+        alertStore.lookupAllUnacknowledged {
+            switch $0 {
+            case .failure(let error):
+                self.log.error("Could not fetch unacknowledged alerts: %@", error.localizedDescription)
+            case .success(let alerts):
+                alerts.forEach { alert in
```

Comment:
Do we really want to replay all of theses alerts all at once? Will that potentially cause overlapping alerts?

Reply:
Yes.  It is not ideal but it is MVP.